Domain/Services/CategoryService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using FluentValidation;
+using LearnNet_CatalogService.Core.DTO;
+using LearnNet_CatalogService.Core.Interfaces;
+using LearnNet_CatalogService.Data.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace LearnNet_CatalogService.Domain.Services
+{
+    public class CategoryService : ICategoryService
+    {
+        private readonly IRepository<Category> _repository;
+        private readonly ILogger<CategoryService> _logger;
+        private readonly IValidator<Category> _categoryValidator;
+
+        public CategoryService(IRepository<Category> repository,
+                               ILogger<CategoryService> logger,
+                               IValidator<Category> categoryValidator)
+        {
+            _repository = repository;
+            _logger = logger;
+            _categoryValidator = categoryValidator;
+        }
+
+        public async Task<bool> AddCategoryAsync(CategoryDTO dto)
+        {
+            var entity = CategoryDTO.MapTo(dto) ?? throw new ArgumentNullException(nameof(dto));
```

Comment:
IMO we should validate CategoryDTO on controller level, so I would remove validation from here

Reply:
I think that there should be two levels of validation: 1 is on controller level with DTO and 2 is on service level with Entity validators.